webui/react/src/components/kit/CodeEditor.tsx

```diff
@@ -31,12 +31,28 @@ const MARKDOWN_CONFIG = {
   highlightActiveLineGutter: false,
 };
 
+type ErrorMessage = {
+  _tag: 'Error';
+  message: string;
+};
+// extension to loadable
+export type LoadableOrError<T> = Loadable<T> | ErrorMessage;
```

Comment:
Maybe leave a TODO here referencing the ticket we have for this. Strong indicator that we should add this into `Loadable`.